pandas/tests/scalar/timestamp/test_timestamp.py

```diff
@@ -1092,3 +1092,17 @@ def test_constructor_ambigous_dst():
     expected = ts.value
     result = Timestamp(ts).value
     assert result == expected
+
+
+def test_constructor_before_dst_switch():
+    # GH 31043
+    # Make sure that calling Timestamp constructor
+    # on time just before DST switch doesn't lead to
+    # nonexistent time
+    epoch = 1552211999999999872
+    ts = Timestamp(epoch, tz="dateutil/US/Pacific")
+    delta = ts.replace(tzinfo=None) - datetime.utcfromtimestamp(0)
+    assert 1552183199.99 < delta.total_seconds() < 1552183200
+    expected = timedelta(seconds=0)
+    result = ts.tz.dst(ts)
+    assert result == expected
```

Comment:
Okay, we now fail only on Linux minimum versions and macOS. Let's fiddle with the implementation a couple more times.